### PR TITLE
Disable `optimize_rewrite_sum_if_to_count_if` ast rewrites for worker nodes in distrubuted queries

### LINKED ISSUE #34862: Do we have some mechanism to ensure that subqueries sent from initial node won't be optimized by rewritten.
**Use case**
Do we have some mechanism to ensure that subqueries sent from initial node won't be optimized by rewritten. 
This would make sense to reduce the exceptions `Not found column` which happened when initial node didn't optimize by rewritting ast but worker node did it.

### PR DESCRIPTION
**Pull request by vdimir**
Changelog category (leave one):

- Not for changelog (changelog entry is not required)

Close https://github.com/ClickHouse/ClickHouse/issues/34862

### CODE REVIEW

**Comment by nickitat** [comment]
I would probably agree with @taiyang-li that turning off the optimizations one-by-one is fragile. IMO it would be ok to let the initiator to made all suitable rewrites and than made peers to not do any rewrite at all (just by a setting). What is your perspective?

**Comment by vdimir** [comment]
Let's disable just `optimize_rewrite_sum_if_to_count_if`. Also I'll try to add test

File: tests/integration/test_backward_compatibility/test_ast_rewrite.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    """)
+
+    nodes[1].query("""
+        INSERT INTO default.table (`reportTime`, `appId`, `platform`, `firstIFrameTs`) VALUES (2, 2, 2, 2);
+    """)
+
+    nodes[1].query("""
+        SELECT
+            toStartOfMinute(toDateTime(reportTime)) AS rtime,
+            sumIf(1, (appId = 60) AND (firstIFrameTs > 0))
```
Comment (nickitat) [inline tests/integration/test_backward_compatibility/test_ast_rewrite.py:68]: idn if it matters but maybe lets add some rows which satisfy the condition?

**Review by nickitat — APPROVED** [review]
overall lgtm

**Comment by vdimir** [comment]
https://github.com/ClickHouse/ClickHouse/issues/34862#issuecomment-1064862644

**Pull request closed without merging**